spec/helpers_spec.rb

```diff
@@ -0,0 +1,191 @@
+      end
+
+      it "for any lower case characters" do
+        expect(downcased_alphabet.chars.any? {|ch| tr_upper?(ch) }).to eq(false)
+      end
+
+      it "for non-letter characters" do
+        expect("\"é1!2'3^+4.,-_".chars.any? {|ch| tr_upper?(ch) }).to eq(false)
+      end
+    end
+  end
+
+  describe "#tr_char?" do
+    it "returns true for any Turkish specific character" do
+      expect(tr_all.chars.all? {|ch| tr_char?(ch) }).to eq(true)
+    end
+
+    context "returns false" do
+      it "for non-Turkish specific characters" do
+        expect(alphabet.delete(tr_all).chars.any? {|ch| tr_char?(ch) }).to eq(false)
+      end
+
+      it "for non-letter characters" do
+        expect("\"é1!2'3^+4.,-_".chars.any? {|ch| tr_char?(ch) }).to eq(false)
+      end
+    end
+  end
+
+  describe "#conjuction?" do
+    it "returns true for any conjuction" do
+      expect(conjuctions.all? {|c| conjuction?(c) }).to eq(true)
+    end
+
+    it "returns false for any word contains conjuction" do
+      expect(%w(veda aile veyahut).any? {|c| conjuction?(c) }).to eq(false)
+    end
+  end
+
+  describe "#start_with_a_special_char?" do
+    it "returns true for all words starts with a special char" do
+      special_words = turkish_words.map { |w| special_chars.sample + w }
+      expect(special_words.all? { |word| start_with_a_special_char?(word) }).to eq(true)
+    end
+
+    it "returns false any words not starts with a special char" do
+      expect(turkish_words.any? { |word| start_with_a_special_char?(word) }).to eq(false)
+    end
+  end
+
+  describe "#translate_range" do
+    it "translates a complete lower-case range correctly" do
+      expect(translate_range('a-z')).to eq(downcased_alphabet)
+    end
+
+    it "translates a complete upper-case range correctly" do
+      expect(translate_range('A-Z')).to eq(upcased_alphabet)
+    end
+
+    it "raises an error if any arguments not a letter" do
+      expect{ translate_range('1-Z') }.to raise_error
+      expect{ translate_range('A-9') }.to raise_error
+      expect{ translate_range('1-9') }.to raise_error
+      expect{ translate_range('a-]') }.to raise_error
+    end
+
+    it "raises an error if arguments are not in same case" do
+      expect{ translate_range('a-Z') }.to raise_error
+    end
+
+    it "translates a partial range correctly" do
+      expect(translate_range('d-t')).to eq('defgğhıijklmnoöpqrsşt')
+      expect(translate_range('Ç-Ü')).to eq('ÇDEFGĞHIİJKLMNOÖPQRSŞTUÜ')
+    end
+
+    it "translates a range  correctly when casefold option is true" do
+      expect(translate_range('d-t', true)).to eq('defgğhıijklmnoöpqrsştĞIİÖŞ')
+      expect(translate_range('C-U', true)).to eq('CÇDEFGĞHIİJKLMNOÖPQRSŞTUçğıiöş')
+    end
+
+    it "translates ranges created using Turkish characters" do
+      expect(translate_range('Ç-Ü', true)).to eq('ÇDEFGĞHIİJKLMNOÖPQRSŞTUÜçğıiöşü')
+      expect(translate_range('ç-ü', true)).to       eq('çdefgğhıijklmnoöpqrsştuüÇĞIİÖŞÜ')
```

Comment:
Line is too long. [89/80]<br>Put one space between the method name and the first argument.